src/core/composer/qgscomposermap.cpp

```diff
@@ -236,6 +236,7 @@ QgsMapSettings QgsComposerMap::mapSettings( const QgsRectangle& extent, QSizeF s
   {
     //if outputing composer, disable optimisations like layer simplification
     jobMapSettings.setFlag( QgsMapSettings::UseRenderingOptimization, false );
+    jobMapSettings.setFlag( QgsMapSettings::RenderPartialOutput, false );
```

Comment:
Shouldn't this be done outside of the plotStyle condition? I.e. in composer maps we'd always disable the partial outputs (since it has no value even in just the preview mode)

Reply:
Agreed, it is better to have it outside the condition. 
My first thought was that the partial output is used in the preview, but that does not seem to be the case.